docs/db.md

```diff
@@ -0,0 +1,48 @@
+# Database Usage
+
+go-exploit supports the use of an SQLite3 database in order to facilite cross-exploit communication and HTTP caching. This is optional, but can greatly improve the performance of large scale scanning. 
+
+To use this feature, use the `-db` command line option. The provided file can be empty or a database created during previous runs of go-exploit. In the example below, we use the option `-db vc.db` to use the non-existent file `vc.db`:
+
+```console
+albinolobster@mournland:~/initial-access/feed/cve-2024-31982$ ls vc.db
+ls: cannot access 'vc.db': No such file or directory
+albinolobster@mournland:~/initial-access/feed/cve-2024-31982$ ./build/cve-2024-31982_linux-arm64 -v -c -rhost 10.9.49.29 -rport 8080 -db vc.db -fll TRACE
+time=2024-06-26T15:54:18.902-04:00 level=DEBUG msg="Using the HTTP User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36"
+time=2024-06-26T15:54:18.911-04:00 level=STATUS msg="Starting target" index=0 host=10.9.49.29 port=8080 ssl=false "ssl auto"=false
+time=2024-06-26T15:54:18.911-04:00 level=STATUS msg="Validating XWiki target" host=10.9.49.29 port=8080
+time=2024-06-26T15:54:19.068-04:00 level=SUCCESS msg="Target verification succeeded!" host=10.9.49.29 port=8080 verified=true
+time=2024-06-26T15:54:19.068-04:00 level=STATUS msg="Running a version check on the remote target" host=10.9.49.29 port=8080
+time=2024-06-26T15:54:19.068-04:00 level=TRACE msg="HTTP cache hit: http://10.9.49.29:8080/"
+time=2024-06-26T15:54:19.069-04:00 level=VERSION msg="The reported version is 14.10.7" host=10.9.49.29 port=8080 version=14.10.7
+time=2024-06-26T15:54:19.070-04:00 level=SUCCESS msg="The target appears to be a vulnerable version!" host=10.9.49.29 port=8080 vulnerable=yes
+albinolobster@mournland:~/initial-access/feed/cve-2024-31982$ ls vc.db 
+vc.db
+```
+
+In the TRACE output we can see that there is an HTTP cache hit during version scanning, this is because the first request in target verfication was cached. If we run the exploit again we will see more TRACE output:
+
+```console
+albinolobster@mournland:~/initial-access/feed/cve-2024-31982$ ./build/cve-2024-31982_linux-arm64 -v -c -rhost 10.9.49.29 -rport 8080 -db vc.db -fll TRACE
```

Comment:
Should we support a write/append-only mode? I could see a situation where you want to continue writing to the cache for multiple checks to keep track of state over time or to have a consistent log of all actions taken. That is less of a cache and more of just a db though.

Reply:
To clarify: do you mean a mode in which the system caches response but doesn't actually use them?

Reply:
Yes that's exactly what I mean.

Reply:
I'm sort of struggling to see the usefulness of the feature. I'm 100% onboard that creating the db is useful for future debugging / logging, but if the data is present in the DB, why not use it?

Reply:
Mostly from an operational stand point if you are wanting traces of all actions was my core thought. Some folks very much want step by step logs of actions for auditability and this sort of provides that functionality without directly having to build it out. But also, if no one is asking /shrug.

Reply:
Sure, I get it. I think my general philosophy is to land the most simple version of a feature as possible, and then to iterate as we want and/or need. I think we are on the same page that an action auditing framework within go-exploit would be great. The db might be the place for that, but I think it's a "next step" type of thing.